### ISSUE TITLE
Predicting political orientation using non-political questionnaire: CNN-based big data survey analysis

### ISSUE THREAD

**Issue by kimsook** [yskim0/One-Day-One-Paper#64]
# 비정치적인 질문으로 정치성향 예측하기: CNN 기반 설문 빅데이터 분석 기법
- The Korean Institute of Information Scientists and Engineers 2020.7, 616-618 (3 pages)

### Abstract
- 비정치적인 설문을 바탕으로 개인의 정치성향을 예측할 수 있는 Convolutional Neural Network (CNN) 기반의 딥러닝 기법 제안
- 학습데이터: 과거 비정치적인 설문에 응답한 설문조사 데이터와 정당에 대한 응답 결과 사용
- 전처리: 설문 조사 데이터를 CNN으로 학습하기 위해 데이터 전처리 기법으로 (1) 설문 빅데이터를 바이너리 이미지화 하여 전처리하는 방법과 (2) 불균형 데이터 문제를 해결하는 설문 데이터 증강 방법 제시

### 학습데이터
- 굿윌헌팅 주식회사 *폴라이드 앱을 통해 수집한 데이터(제공받음)
- 그 중, 2018년 전국동시지방선거 당시 올라온 "내가 지지하는 정당은?" 이라는 질문에 답한 1223명에 대한 설문 응답 데이터 사용
*폴라이드는 잠금화면 리워드 앱으로 사용자가 관심분야를 선택하여 4지 선다형 질문을 올리면 다른 사용자들이 해당 질문에 답을 하고 리워드를 얻는 서비스
- 과거 사용자가 응답한 분야별 설문 데이터
![image](https://user-images.githubusercontent.com/40443049/129449852-8690fa4f-e8be-4e30-93ec-7985eb9ed29c.png)
- 비정치적 질문에 대한 데이터만 학스하기 위해 위 리스트 중 정치와 관련된 카테고리 및 질문을 삭제

### 데이터 전처리
- 설문결과를 이미지화 하는 방식으로 수행
- 행은 문제 카테고리, 열은 문제별 응답결과로 나타냄
- 원-핫 인코딩을 사용하여 4지선다형의 1~4번을 1000,0100, 0010, 0001 로 미응답을 0000으로 변환
- 데이터 증강 수행 --> 응답한 질문의  10% 문제를 랜덤으로 삭제
- 불균형 했던 _정당 선택 클래스_의 비율을 25%씩으로 동등하게 맞추어 학습 진행
*정당 선택 클래스는 당시 정당지지에 대한 분포로 분류 라벨로 사용

### CNN 학습
![image](https://user-images.githubusercontent.com/40443049/129450079-eb6a0796-0924-44d5-be9c-58821a0bf718.png)
- 8:2로 train과 test 진행하며 train 데이터에 데이터 증강을 진행하여 총 11938개 학습
- padding을 same으로 두어 정보 손실이 없도록 함
- 1st fully connected layer에서 나이, 직어, 성별이 concatenate 되도록 함
- 4개의 정당 분류를 위해 softmax 함수 사용
- 1X3 커널 사용, max pooling은 1X2 사용, stride 는 2를 적용

### 결론
- accuracy: 51.09%
- 기존 응답 이력을 바탕으로 필터링된 우수 사용자 데이터가 필요함

**Comment by HYUNMIN-HWANG** [comment]
우와 설문지 내용을 이미지화해서 CNN으로 훈련시킨다니 너무 신기해요!! 근데 acc가 낮군요ㅠㅠ
